warn/warn_docstring.go

```diff
@@ -0,0 +1,250 @@
+	if f.Type != build.TypeDefault {
+		return []*Finding{}
+	}
+	if stmt, ok := getDocstring(f.Stmt); stmt != nil && !ok {
+		start, end := stmt.Span()
+		return []*Finding{
+			makeFinding(f, start, end, "module-docstring",
+				`The file has no module docstring.`, true, nil),
+		}
+	}
+	return []*Finding{}
+}
+
+func stmtsCount(stmts []build.Expr) int {
+	result := 0
+	for _, stmt := range stmts {
+		result++
+		switch stmt := stmt.(type) {
+		case *build.IfStmt:
+			result += stmtsCount(stmt.True)
+			result += stmtsCount(stmt.False)
+		case *build.ForStmt:
+			result += stmtsCount(stmt.Body)
+		}
+	}
+	return result
+}
+
+// docstringInfo contains information about a function docstring
+type docstringInfo struct {
+	hasHeader     bool            // whether the docstring has a one-line header
+	args          map[string]bool // map of documented arguments
+	returns       bool            // whether the return value is documented
+	argumentsLine int             // line of the `Arguments:` block (not `Args:`), if it exists
+}
+
+// docstringBlock contains a block of a docstring (separated by empty lines)
+type docstringBlock struct {
+	startLineNo int      // line number of the first line of the block
+	lines       []string // lines
+}
+
+// countLeadingSpaces returns the number of leading spaces of a string.
+func countLeadingSpaces(s string) int {
+	spaces := 0
+	for _, c := range s {
+		if c == ' ' {
+			spaces++
+		} else {
+			break
+		}
+	}
+	return spaces
+}
+
+var argHeaderRegex = regexp.MustCompile("^ *(\\*?\\*?\\w+)( *\\([\\w\\ ,]+\\))?:")
+var argRegex = regexp.MustCompile("[\\*\\w]+")
+
+// parseFunctionDocstring parses a function docstring and returns a docstringInfo object containing
+// the parsed information about the function, its arguments and its return value
+func parseFunctionDocstring(doc *build.StringExpr) docstringInfo {
+	start, _ := doc.Span()
+	indent := start.LineRune - 1
+	lines := strings.Split(doc.Value, "\n")
+
+	// Dedent the lines
+	for i, line := range lines {
+		line = strings.TrimRight(line, " ")
+		if i != 0 {
+			for j, chr := range line {
+				if j >= indent || chr != ' ' {
+					line = line[j:]
+					break
+				}
+			}
+		}
+		lines[i] = line
+	}
+
+	// Split by empty lines
+	blocks := []docstringBlock{}
+	newBlock := true
+	for i, line := range lines {
+		if len(line) == 0 {
+			newBlock = true
+			continue
+		}
+		if newBlock {
+			newBlock = false
+			blocks = append(blocks, docstringBlock{
+				startLineNo: start.Line + i,
+				lines:       []string{},
+			})
+		}
+		blocks[len(blocks)-1].lines = append(blocks[len(blocks)-1].lines, line)
+	}
+
+	info := docstringInfo{}
+	info.args = make(map[string]bool)
+
+	if len(blocks) > 0 && len(blocks[0].lines) == 1 {
+		// Exactly one line in the first block
+		info.hasHeader = true
+	}
+
+	// Iterate over the blocks, extract data
+	for _, block := range blocks {
+		switch block.lines[0] {
+		case "Args:", "Arguments:":
+			if block.lines[0] == "Arguments:" {
```

Comment:
Add comment, to state why you look at "Arguments"

Reply:
Done.